File: convertC2Nim/cmonte.nim
## http://www.sat.t.u-tokyo.ac.jp/~omi/random_variables_generation.html#Uniform

proc UniformRandom01*(): cdouble =
  return (cast[cdouble](rand()) + 1.0) div (cast[cdouble](RAND_MAX) + 2.0)

proc main*(): cint =
  srand(time(nil))
  var N: cint = 100000000
  var counter: cint = 0
  var i: cint = 0
  while i < N:
    var x: cdouble = UniformRandom01()
    var y: cdouble = UniformRandom01()
    if x * x + y * y < 1.0:
      inc(counter, 1)
    inc(i)
  printf("approx pi is %f\x0A", 4 * counter div cast[cdouble](N))

File: convertC2Nim/cmonte.c
#include <time.h>
#include <stdio.h>
#include <stdlib.h>

//http://www.sat.t.u-tokyo.ac.jp/~omi/random_variables_generation.html#Uniform
double UniformRandom01(void){
    return ((double)rand()+1.0)/((double)RAND_MAX+2.0);
}
int main(void){
    srand(time(NULL));
    const int N=100000000;
    int counter=0;
    for (int i=0;i<N;++i){
        double x=UniformRandom01();
        double y=UniformRandom01();
        if (x*x+y*y<1.0){
            counter+=1;
        }
    }
    printf("approx pi is %f\n", 4*counter/(double)N);
}